File: nomad-samples/js/asset-upload/assets/upload-asset-part.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

export default async function uploadAssetPart(AUTH_TOKEN, PART_ID, ETAG) 
{
    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${AUTH_TOKEN}`);
  
    const BODY = {
        etag: ETAG
    };

    // Post
    const RESPONSE = await fetch(`${prjConstants.ADMIN_API_URL}/asset/upload/part/${PART_ID}/complete`, {
        method: "POST",
        headers: HEADERS,
        body: JSON.stringify(BODY)
    }).catch((exception) => {
        throw exception;
    });

    // Check for success
    if (RESPONSE.ok) {
        // Get the JSON from the response
        const INFO = await RESPONSE.json();

        return INFO;
    }
		
  	// There was an error
    await apiExceptionHandler(RESPONSE, "Upload asset part failed");
}

File: nomad-samples/common/py/asset_upload/assets/upload_asset_part.py
from exceptions.api_exception_handler import *

import requests

def upload_asset_part(FILE, URL, PART):
    if not FILE or not URL or not PART:
        raise Exception("Upload Part: Invalid API call")
    
    try:
        OPEN_FILE = open(FILE, "rb")
        OPEN_FILE.seek(PART["startingPosition"])
        BODY = OPEN_FILE.read(PART["endingPosition"] + 1 - PART["startingPosition"])
        OPEN_FILE.close()


        HEADER = {
            "Accept": "application/json, text/plain, */*"
        }

        RESPONSE = requests.put(URL, headers=HEADER, data=BODY)

        if not RESPONSE.ok:
            raise Exception()
            
        return RESPONSE.headers.get("ETag")
    
    except:
        api_exception_handler(RESPONSE, "Upload part failed")